### ISSUE TITLE
Wrong colors and bugged log output (BB #754)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#754]
This issue was migrated from bitbucket.
**Original Reporter:** PadreWoW
**Original Date:** 06.12.2014 08:06:15 GMT+0000
**Original Priority:** minor
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754

<hr>

Last rev without any custom patches

Colors are wrong

LogColors = "9 6 2 2" - 9 - LRED, 6 - CYAN

Errors are BLUE - not red and I can't change it

Output is still mixed (no line breaks)

![2001035143-1.jpg](https://bitbucket.org/repo/kMeMB/images/3474401597-2001035143-1.jpg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 06.12.2014 08:06:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-14115005

<hr>

Issue #702 was marked as a duplicate of this issue.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 09.12.2014 15:40:39 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-14175493

<hr>

I think the problem with colors is here:

```
#!c++

for (i = 0; i < len - 1; i++)
        {
            switch (buffer[i])
            {
            case 0xFF:
                control = 1;
                continue;
            case 0xFE:
                wattrset(m_logViewer, A_NORMAL);
                continue;
            default:
                if (control)
                {
                    control = 0;
                    if (buffer[i] > 7)
                        wattrset(m_logViewer, A_BOLD | TermColor(buffer[i] - 8));
                    else
                        wattrset(m_logViewer, TermColor(buffer[i]));
                    continue;
                }
                waddch(m_logViewer, buffer[i]);
            }
        }
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 10.12.2014 08:51:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-14190485

<hr>

In config:

LogLevel = 1

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 10.12.2014 09:37:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-14191440

<hr>

What the hell is "buffer[i]" and what is "control"?

Platform: Windows Server

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 04.02.2015 04:28:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-15459380

<hr>

Tested, with exact same config options. Colors were outputted correctly.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 04.02.2015 06:39:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-15460875

<hr>

Do you use Linux?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** ShanduLIVE
**Original Date:** 04.02.2015 09:11:49 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-15464088

<hr>

May I ask how I get such a console view? I press enter but see any Log Infos.
I already have "LogLevel = 1" and for example he dont make GM_.log too

Logs/GMs/GM_Commands.log , is used.

greetZ
- Shandu

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** desteny
**Original Date:** 04.02.2015 15:28:09 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-15472864

<hr>

you may need a higher log level (2 or 3) for detailed or debug view

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 04.02.2015 17:27:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-15476335

<hr>

I use Windows 7. Console.Enable set to 1. Colors set to "9 6 2 2".
http://imgur.com/3PyC0KP

I don't see the issue. Can someone point it out?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 18.02.2015 13:47:19 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-15771809

<hr>

Colors are apparently working, and about the broken line-breaking, it'll be fixed in no time.

![colors.jpeg](https://bitbucket.org/repo/kMeMB/images/651307358-colors.jpeg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 18.02.2015 13:47:27 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/754/#comment-15771811

<hr>

Fix #754

→ <<cset f5ad584452e2>>
